### PR TITLE
feat: add Button.shape 'default'

### PR DESCRIPTION
**Pull request by jaredleechn**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Add Button `shape` `default` ts definiton.       |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by zombieJ** [comment]
CI failed

**Comment by jaredleechn** [comment]
It's confusing that Button get same classNames `ant-btn-default` from props.shape and props.type

How about use null or 'radius' instead?

File: components/button/button.tsx [hunk 1]
```diff
@@ -241,7 +242,7 @@ const InternalButton: React.ForwardRefRenderFunction<unknown, ButtonProps> = (pr
     prefixCls,
     {
       [`${prefixCls}-${type}`]: type,
-      [`${prefixCls}-${shape}`]: shape,
+      ...(shape && shape !== 'default' && { [`${prefixCls}-${shape}`]: true }),
```
Comment (zombieJ) [inline components/button/button.tsx:245]:
```suggestion
      [`${prefixCls}-${shape}`]: shape !== 'default' && shape,
```

Comment (jaredleechn) [reply components/button/button.tsx:245]: 不行，之前就是这么写的，遇到这个 case 会挂

```
<Button type="default" shape="default" />

classnames({
  ant-btn-default: true,
  ant-btn-default: false,
})
```
就把 type default 的给覆盖掉了，我还尝试了 classnames(a, b) 这样的写法也不行

**Pull request merged**